### PR TITLE
fix: fix RestoreStreamer to prevent buckets skipping #2830

### PR DESCRIPTION
**Pull request by BorysTheDev**

### CODE REVIEW

File: tests/dragonfly/cluster_test.py [hunk 1]
```diff
@@ -1177,11 +1177,6 @@ async def test_cluster_fuzzymigration(
     seeder = df_seeder_factory.create(keys=keys, port=nodes[0].instance.port, cluster_mode=True)
     await seeder.run(target_deviation=0.1)
 
-    fill_task = asyncio.create_task(seeder.run())
```
Comment (adiholden) [inline tests/dragonfly/cluster_test.py:1180]: why do you remove this?

File: src/server/journal/streamer.cc [hunk 2]
```diff
@@ -83,6 +83,15 @@ void RestoreStreamer::Start(io::Sink* dest, bool send_lsn) {
...
     bool written = false;
     cursor = pt->Traverse(cursor, [&](PrimeTable::bucket_iterator it) {
+      uint64_t v = it.GetVersion();
+      if (v >= snapshot_version_) {
+        // either has been already serialized or added after snapshotting started.
+        DVLOG(3) << "Skipped " << it.segment_id() << ":" << it.bucket_id() << ":" << it.slot_id()
+                 << " at " << v;
+      }
+      constexpr DbIndex cluster_db_index = 0;
+      db_slice_->FlushChangeToEarlierCallbacks(cluster_db_index, DbSlice::Iterator::FromPrime(it),
```
Comment (adiholden) [inline src/server/journal/streamer.cc:93]: nit:  FlushChangeToEarlierCallbacks(0 /*db_id*/, ...)

File: tests/dragonfly/cluster_test.py (hunk 1 above)
Comment (BorysTheDev) [reply tests/dragonfly/cluster_test.py:1180]: I don't see any point in having 2 calls for Seeder

Comment (dranikpg) [reply tests/dragonfly/cluster_test.py:1180]: I didn't track the changes to this part of code, but there used to be a reason, likely now removed or just moved... Almost sure 😆

File: src/server/journal/streamer.cc [hunk 3]
```diff
@@ -83,6 +83,15 @@ void RestoreStreamer::Start(io::Sink* dest, bool send_lsn) {
 
     bool written = false;
     cursor = pt->Traverse(cursor, [&](PrimeTable::bucket_iterator it) {
+      uint64_t v = it.GetVersion();
+      if (v >= snapshot_version_) {
+        // either has been already serialized or added after snapshotting started.
+        DVLOG(3) << "Skipped " << it.segment_id() << ":" << it.bucket_id() << ":" << it.slot_id()
```
[the remark this answers is not in the document]
Comment (BorysTheDev) [reply src/server/journal/streamer.cc:89]: I decided to remove at all

File: tests/dragonfly/cluster_test.py (hunk 1 above)
Comment (adiholden) [reply tests/dragonfly/cluster_test.py:1180]: Ok now I see that the seeder is not running while we are doing migration, right?
why so?

Comment (BorysTheDev) [reply tests/dragonfly/cluster_test.py:1180]: we didn't do it previously too, during migration, we run counters.

Comment (BorysTheDev) [reply tests/dragonfly/cluster_test.py:1180]: we don't control now when migration is finished so seeder can't be stopped before we migration is finished

**Review by adiholden — APPROVED** [review]

**Pull request merged**